lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -126,32 +126,76 @@ def _proc_upload_reads_params(self, params):
-                      'read_orientation_outward': read_orientation_out
-                      })
+
+        source_reads_ref = params.get('source_reads_ref')
+        if source_reads_ref:
+            # Means the uploaded reads is a result of an input reads object being filtered/trimmed
+            try:
+                source_reads_object = dfu.get_objects({'object_refs':
+                                                       [source_reads_ref]})['data'][0]
+            except DFUError as e:
+                self.log(('The supplied source_reads_ref {} was not able to be retrieved. ' +
+                          'Logging stacktrace from workspace exception:' +
+                          '\n{}').format(source_reads_ref, e.data))
+                raise
+            # Make sure that no non file related parameters are set. If so throw error.
+            parameters_should_unfilled = ['insert_size_mean', 'insert_size_std_dev',
+                                          'sequencing_tech', 'strain',
+                                          'source', 'read_orientation_outward']
+            if any(x in params for x in parameters_should_unfilled):
+                self.log(("'source_reads_ref' was passed, making the following list of " +
+                          "parameters {} erroneous to " +
+                          "include").format(",".join(parameters_should_unfilled)))
+                raise ValueError(("'source_reads_ref' was passed, making the following list of " +
+                                  "parameters : {} erroneous to " +
+                                  "include").format(", ".join(parameters_should_unfilled)))
+            # Check that it is a reads object. If not throw an eror.
+            single_input, kbasefile = self.check_reads(source_reads_object)
+            o = {'sequencing_tech': source_reads_object['data'].get("sequencing_tech"),
+                 'single_genome': source_reads_object['data'].get("single_genome")
+                 }
+            if 'strain' in source_reads_object['data']:
+                o['strain'] = source_reads_object['data']['strain']
+            if 'source' in source_reads_object['data']:
+                o['source'] = source_reads_object['data']['source']
+            if not single_input and not single_end:
+                # is a paired end input and trying to upload a filtered/trimmed
+                # paired end ReadsUtils. need to check for more fields.
+                ism = source_reads_object['data'].get('insert_size_mean')
+                issd = source_reads_object['data'].get('insert_size_std_dev')
+                read_orientation_out = source_reads_object['data'].get('read_orientation_outward')
+                o.update({'insert_size_mean': ism,
+                          'insert_size_std_dev': issd,
+                          'interleaved': interleaved,
+                          'read_orientation_outward': read_orientation_out
+                          })
```

Comment:
I still think this is a good idea to do - the code setting up the new object is very similar in both cases.